crate_universe/src/utils/starlark/glob.rs

```diff
@@ -23,23 +22,3 @@ impl Glob {
         // there are no included globs. A glob cannot have only excludes.
     }
 }
-
-impl Serialize for Glob {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        if self.exclude.is_empty() {
-            // Serialize as glob([...]).
```

Comment:
@dtolnay Hey, I know you wanted to try and render cleaner files but seems this breaks deserialization. Do you feel strongly about this functionality? I'm happy to make changes here but do want to get a fix merged and a `0.16.1` released

Reply:
I do prefer `glob([...])` over `glob(include = [...], exclude=[])` if we can make it work. 3 possible solutions are:

1. Use two different `Glob` types between src/context/crate_context.rs and src/utils/starlark.rs (i.e. the opposite of https://github.com/bazelbuild/rules_rust/pull/1734#discussion_r1059472286)
2. Use the same `Glob` type but with a `serialize_with` attribute on either the crate_context.rs side or starlark.rs side to plug in different serialized representation for it
3. Use the same `Glob` type and the same `Serialize` implementation but write a more advanced `Deserialize` impl for `Glob` that can correctly deserialize both JSON `[...]` and `{"include":[...],"exclude":[...]}`

Reply:
I think option 3 sounds like the better one. Would you have time to write up a patch? It may take me a bit.

Reply:
1. a21dedec26e146929f0401ab5f5e50e5746e4a9e
2. efca305d33634e3e410e8ed86610cf3cb0d77df2 or b7a94abd1fafd3b287ba623f7e64718c5090904b
3. bbad25f750a91806458199c66199bdc676cc14ff

Reply:
The implementation (3) from https://github.com/bazelbuild/rules_rust/pull/1763 has been added to this PR.